tests/unit_tests/llms/test_chatglm.py

```diff
@@ -1,61 +1,62 @@
-from llmlite.llms.chatglm2 import ChatGLMChat
-from llmlite.llms.chat import ASSISTANT_PROMPT, SYSTEM_PROMPT, USER_PROMPT
+from llmlite.llms.chatglm2 import ChatGLMChat, build_history
+from llmlite.llms.chat import ASSISTANT_PROMPT, USER_PROMPT
 from llmlite.llms.messages import ChatMessage
 
 
 class TestChatGLM:
-    def test_prompt(self):
+    def test_build_history(self):
         test_cases = [
             {
-                "name": "prompt with both system_prompt and user_prompt",
-                "messages": [
-                    ChatMessage(role=SYSTEM_PROMPT, content="你是一个聪明的机器人"),
-                    ChatMessage(role=USER_PROMPT, content="你是谁"),
-                ],
-                "expected": """问：你是谁
-""",
-            },
-            {
-                "name": "prompt with both system_prompt, user_prompt and assistant_prompt",
                 "messages": [
-                    ChatMessage(role=SYSTEM_PROMPT, content="你是一个聪明的机器人"),
                     ChatMessage(role=USER_PROMPT, content="你是谁"),
-                    ChatMessage(role=ASSISTANT_PROMPT, content="我是一个智能机器人"),
                 ],
-                "expected": """问：你是谁
-答：我是一个智能机器人
-""",
+                "want": ("你是谁", []),
             },
             {
-                "name": "continuous conversation with system_prompt exists",
                 "messages": [
-                    ChatMessage(role=SYSTEM_PROMPT, content="你是一个聪明的机器人"),
                     ChatMessage(role=USER_PROMPT, content="你是谁"),
-                    ChatMessage(role=ASSISTANT_PROMPT, content="我是一个智能机器人"),
-                    ChatMessage(role=USER_PROMPT, content="你会做什么"),
+                    ChatMessage(role=ASSISTANT_PROMPT, content="我是一个聪明的机器人"),
+                    ChatMessage(role=USER_PROMPT, content="你可以帮我做作业吗"),
                 ],
-                "expected": """问：你是谁
-答：我是一个智能机器人
-问：你会做什么
-""",
+                "want": ("你可以帮我做作业吗", [("你是谁", "我是一个聪明的机器人")]),
             },
+        ]
+
+        for tc in test_cases:
+            got = build_history(tc["messages"])
+            assert got == tc["want"]
```

Comment:
**Suggestion:** Consider using pytest.mark.parametrize to avoid code duplication in your tests.
```suggestion
    @pytest.mark.parametrize(
        "messages, expected",
        [
            (
                [
                    ChatMessage(role=USER_PROMPT, content="你是谁"),
                ],
                ("你是谁", []),
            ),
            (
                [
                    ChatMessage(role=USER_PROMPT, content="你是谁"),
                    ChatMessage(role=ASSISTANT_PROMPT, content="我是一个聪明的机器人"),
                    ChatMessage(role=USER_PROMPT, content="你可以帮我做作业吗"),
                ],
                ("你可以帮我做作业吗", [("你是谁", "我是一个聪明的机器人")]),
            ),
        ],
    )
    def test_build_history(self, messages, expected):
        got = build_history(messages)
        assert got == expected
```